integrations/amazon_bedrock/tests/test_chat_generator.py

```diff
@@ -55,128 +52,242 @@ def tools():
+                        "generation_kwargs": {"temperature": 0.7},
+                        "stop_words": ["eviscerate"],
+                        "streaming_callback": "haystack.components.generators.utils.print_streaming_chunk",
+                        "boto3_config": None,
+                        "tools": [
+                            {
+                                "type": "haystack.tools.tool.Tool",
+                                "data": {
+                                    "name": "weather",
+                                    "description": "useful to determine the weather in a given location",
+                                    "parameters": {"city": {"type": "string"}},
+                                    "function": "tests.test_chat_generator.weather",
+                                },
+                            }
+                        ],
+                    },
+                }
+            },
+            "connections": [],
+        }
 
-def test_constructor_with_empty_model():
-    """
-    Test that the constructor raises an error when the model is empty
-    """
-    with pytest.raises(ValueError, match="cannot be None or empty string"):
-        AmazonBedrockChatGenerator(model="")
+        if not hasattr(pipeline, "_connection_type_validation"):
+            expected_dict.pop("connection_type_validation")
+
+        # add outputs_to_string, inputs_from_state and outputs_to_state tool parameters for compatibility with
+        # haystack-ai>=2.12.0
+        if hasattr(tool, "outputs_to_string"):
+            expected_dict["components"]["generator"]["init_parameters"]["tools"][0]["data"][
+                "outputs_to_string"
+            ] = tool.outputs_to_string
+        if hasattr(tool, "inputs_from_state"):
+            expected_dict["components"]["generator"]["init_parameters"]["tools"][0]["data"][
+                "inputs_from_state"
+            ] = tool.inputs_from_state
+        if hasattr(tool, "outputs_to_state"):
+            expected_dict["components"]["generator"]["init_parameters"]["tools"][0]["data"][
+                "outputs_to_state"
+            ] = tool.outputs_to_state
+
+        assert pipeline_dict == expected_dict
+
+    def test_prepare_request_params_tool_config(self, top_song_tool_config, mock_boto3_session, set_env_variables):
+        generator = AmazonBedrockChatGenerator(model="anthropic.claude-3-5-sonnet-20240620-v1:0")
+        request_params, callback = generator._prepare_request_params(
+            messages=[ChatMessage.from_user("What's the capital of France?")],
+            generation_kwargs={"toolConfig": top_song_tool_config},
+            tools=None,
+        )
+        assert request_params["messages"] == [{"content": [{"text": "What's the capital of France?"}], "role": "user"}]
+        assert request_params["toolConfig"] == top_song_tool_config
 
 
 class TestAmazonBedrockChatGeneratorInference:
```

Comment:
if every test case in this class is an integration test, you can apply the following decorators to the class:
- `@pytest.mark.integration`
- `@pytest.mark.skipif`